gpii/node_modules/rawPreferencesServer/src/rawPreferencesGet.js

```diff
@@ -25,18 +25,31 @@ https://github.com/GPII/universal/blob/master/LICENSE.txt
         invokers: {
             handle: {
                 funcName: "gpii.handleRawPreferencesGet",
-                args: ["{requestProxy}", "{request}.req.params.userToken", "{rawPreferencesServer}.rawPreferencesSource"],
+                args: [ "{request}", "{request}.req.params.userToken", "{rawPreferencesServer}.rawPreferencesSource"],
                 dynamic: true
             }
         }
     });
 
-    gpii.handleRawPreferencesGet = function (requestProxy, userToken, rawPreferencesSource) {
-        rawPreferencesSource.get({
-            userToken: userToken
-        }, function (preferences) {
-            gpii.rawPreferencesServer.handlePreferences(requestProxy, userToken, preferences);
+    gpii.handleRawPreferencesGet = function (request, userToken, rawPreferencesSource) {
+        var promise = rawPreferencesSource.get({userToken: userToken});
+        promise.then(function (preferences) {
+            if (preferences === undefined) {
+                preferences = {
+                    isError: true,
+                    message: "Unable to retrieve raw preferences for user " + userToken,
+                    statusCode: 404
+                };
+            }
+            gpii.rawPreferencesServer.handlePreferences(request, userToken, preferences);
+        }, function (err) {
+            // slightly odd "err.message || err.error" here due things like the parser sending their output
+            // in an 'error' key
+            request.events.onError.fire({
+                isError: true,
+                message: "Rejected promise from raw preferences server.. Reason: " + (err.message || err.error),
```

Comment:
"Rejected promise" is a peculiarly technical expression to put in an error message which may in some cases even appear on the user's screen : P "Error" will be just as accurate (since promises don't actually exist in the real (HTTP) world) and more intelligible